pkg/storage/external/client.go

```diff
@@ -0,0 +1,183 @@
+package external
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"strings"
+
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/storage"
+	"golang.org/x/mod/module"
+)
+
+type service struct {
+	url string
+	c   *http.Client
+}
+
+// NewClient returns an external storage client
+func NewClient(url string, c *http.Client) storage.Backend {
+	if c == nil {
+		c = http.DefaultClient
+	}
```

Comment:
Can you force a specific client, so we are not using the default? It's a well-configured client but I'm always concerned that some other part of Athens will use it later and there will be connection contention